tutorials/15-external-samplers/external-samplers.jmd

```diff
@@ -0,0 +1,104 @@
+---
+title: Using External Sampler
+permalink: /tutorials/:name/
+mathjax: true
+weave_options:
+  error : false
+---
+
+# Using External Samplers
+
+Turing currently wraps some of the most popular sampling libraries in the Julia ecosystem.
+These wrappers allow new users to seemlessly sample statistical models without leaving Turing.
+However, these wrappers often lack the latests functionalities implemented on the sampling libraries that more experienced users might want to use.
+Moreover, users might want to use samplers currently not wrapped within Turing.
+
+For these reasons, Turing now allows users the possibility of defining their own sampler and using it to sample a Turing model.
+Let us begin by defining a simple Turing model to showcase the use of external samplers.
+For this demonstration we will use a 10 dimensional Neal's funnel:
+
+```julia
+# Import libraries.
+using Turing, Random, LinearAlgebra
+
+d = 10
+@model function funnel()
+    θ ~ Truncated(Normal(0, 3), -3, 3)
+    z ~ MvNormal(zeros(d - 1), exp(θ) * I)
+    return x ~ MvNormal(z, I)
+end
+```
+
+Let us know sample the model to generate some observations and condition our model on them:
+
+```julia
+Random.seed!(1)
+(; x) = rand(funnel() | (θ=0,))
+model = funnel() | (; x)
+```
+
+Users can now use any sampler algorithm they wish to sample this model as long as it is of the type `AbstractMCMC.AbstractSampler`.
+Before discussing how is this in done in practice, it is interesting to give a high level description of the process.
+Imagine that we created an instance of external sampler that we will call `spl` such that `typeof(spl)<:AbstractMCMC.AbstractSampler`.
+In order to avoid type ambiguity within Turing, at the moment is necessary to declare `spl` as an external sampler to Turing this is done as:
+
+```julia
+espl = externalsampler(spl)
+```
+
+where `externalsampler(s::AbstractMCMC.AbstractSampler)` is a Turing function that types our external sampler adquately.
+
+Arrived at this point sampling our model is as easy as:
+
+```julia
+chain = sample(model, espl, 1_000)
+```
+
+This process will use `AbstractMCMC` functions overloaded by Turing to generate chains identical to those generated by the native samplers.
+Note that if the external sampler produces transitions that Turing cannot parse the bundling of the samples will be different or fail.
```

Comment:
IMO this would do well with further elaboration; might even warrant it's own section, preferably with a simple example, as this is likely to be a source of confusion for the user.

Reply:
I am afraid I don't know enough about the bundling process to know what's important information to transmit to the reader. 
I also think that we can remove this sentence since user level readers will not interact with this and devs only really need to know how to define their transition object in their libraries which is covered later. 
However, please feel free to fill in information you think it might be useful.